pilot/pkg/xds/endpoint_builder.go

```diff
@@ -88,6 +90,8 @@ func NewEndpointBuilder(clusterName string, proxy *model.Proxy, push *model.Push
 		port:       port,
 	}
 
+	b.populateFailoverPriorityLabels()
```

Comment:
I am not sure why we need to convert this to bytes and then later also convert it to hash. Why not just directly go to hash?

Reply:
If we convert this to hash, what would we do in `Key()`? Do you mean hashing it again in `Key()`?

Reply:
This is good, I am just concerned if the failover priority labels are very customized, the scale of the cache could be per proxy in the worst case.

Do we need to only cache some with standard labels, like Netwok. Cluster, Region/zone/subzone

Reply:
> Do we need to only cache some with standard labels, like Netwok. Cluster, Region/zone/subzone

I'm not sure, @howardjohn WDYT

Reply:
I think we need all of the ones referenced? otherwise if they select more it won't work?

Reply:
Since we have cache size limit, so looks not a big matter. Will trigger evict if the keys number is  too large